src/core/app.lua

```diff
@@ -119,66 +122,134 @@ end
 -- new app network by making the changes needed.
 function configure (new_config)
    local actions = compute_config_actions(configuration, new_config)
-   apply_config_actions(actions, new_config)
-   configuration = new_config
```

Comment:
I am not sure that what you are working on is a good idea.  it does not seem to be a strategy that will survive a multiprocess world.